zilliqa/src/transaction.rs

```diff
@@ -557,6 +567,20 @@ impl FromStr for ScillaGas {
 #[serde(transparent)]
 pub struct EvmGas(pub u64);
 
+impl EvmGas {
+    pub fn checked_sub(self, rhs: EvmGas) -> Option<EvmGas> {
+        Some(EvmGas(self.0.checked_sub(rhs.0)?))
+    }
+}
+
+impl Sub for EvmGas {
+    type Output = EvmGas;
+
+    fn sub(self, rhs: Self) -> Self::Output {
+        self.checked_sub(rhs).expect("evm gas underflow")
```

Comment:
What happens if I can contrive for this to panic?

Reply:
Bad things. We catch panics in the context of API calls, but not when processing messages. I've raised #1004 to fix this.

Once merged, it will mean this node fails to propose a block and the network new views. Hopefully the same doesn't occur again, but again.. It depends on what the bug is :)